stable/helm-exporter/templates/service.yaml

```diff
@@ -13,7 +13,9 @@ metadata:
   {{- end }}
 spec:
   type: ClusterIP
+  {{- if not .Values.ingress.enabled }}
```

Comment:
With this change can someone turn this flag on and off or will it consider this immutable?

Reply:
Tested it out. It will consider it immutable.

Firstly I installed using `ingress.enabled` as `false`

Then I set `ingress.enabled` as `true`
```
$ helm upgrade helm-exporter-test .
UPGRADE FAILED
ROLLING BACK
Error: Service "helm-exporter-test" is invalid: spec.clusterIP: Invalid value: "": field is immutable
Error: UPGRADE FAILED: Service "helm-exporter-test" is invalid: spec.clusterIP: Invalid value: "": field is immutable
```

Reply:
Also if one were to upgrade their helm chart using this as the new version and they haven't updated `values.yaml` to match the example in this PR then they would get the error

`Error: render error in "helm-exporter/templates/service.yaml": template: helm-exporter/templates/service.yaml:16:20: executing "helm-exporter/templates/service.yaml" at <.Values.ingress.enab...>: can't evaluate field enabled in type interface {}`

Not sure how we would like it handled. This would affect those who upgrade chart but perhaps use their own `values.yaml` file and don't look at the example. Did a bit of looking around but doesn't seem to be a nice way of doing it? https://github.com/helm/helm/issues/5180

Reply:
Not following what you mean by the example?  The `values.yaml` here is embedded into the chart.

Reply:
Is none required here by ingress?

Reply:
Actually regarding None. Seems to work fine with ingress, I was under the impression it might not work as expected. I will revert that part

Yes I mean the `values.yaml` inbedded into the chart. Simply put not having the `ingress` section in your `values.yaml` will not work, which is why it is updated here in the PR. I'm just saying if I was someone who just does a `helm repo update` then `helm install ... stable/helm-exporter` then it would not work for them until they add a similar section into the `values.yaml` file

```
ingress
  enabled: false

  annotations: {}
  path: /
  host:
```